src/LiveComponent/src/Metadata/LiveComponentMetadataFactory.php

```diff
@@ -33,12 +33,17 @@ public function __construct(
 
     public function getMetadata(string $name): LiveComponentMetadata
     {
+        static $liveComponentMetadata;
+        if (isset($liveComponentMetadata[$name])) {
+            return $liveComponentMetadata[$name];
+        }
+
         $componentMetadata = $this->componentFactory->metadataFor($name);
 
         $reflectionClass = new \ReflectionClass($componentMetadata->getClass());
         $livePropsMetadata = $this->createPropMetadatas($reflectionClass);
 
-        return new LiveComponentMetadata($componentMetadata, $livePropsMetadata);
+        return $liveComponentMetadata[$name] = new LiveComponentMetadata($componentMetadata, $livePropsMetadata);
```

Comment:
Are we sure we want to share a single metadata instance for all the components of the same name ?

Reply:
The ComponentFactory metadataFor method also returns ComponentMetadata with configurations based on name, so it won't be a problem.

Reply:
But is the instance modified elsewhere in the code ?

Reply:
The metadata DTO `LiveComponentMetadata` that will be stored only has private propeties and getters. All data within the DTO is static and related to property attribute `LiveProp` and static component configurations.

Reply:
The metadata DTO `LiveComponentMetadata` that will be stored only has private propeties and getters. All data within the DTO is static and related to property attribute `LiveProp` and static component configurations.

Reply:
Yup, this looks good to me too. `LiveComponentMetadata` is meant to be the metadata for the component with the name `$name`. If we ever modify any of that metadata on component-by-component basis, that should be considered a bug (and I don't see any of that at the moment).